controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    velocity_controller1:
      type: velocity_controllers/JointGroupVelocityController
    
    velocity_controller2:
      type: velocity_controllers/JointGroupVelocityController

    position_controller:
      type: position_controllers/JointGroupPositionController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

velocity_controller1:
  ros__parameters:
    joints:
      - leg1_joint
    interface_name: velocity
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity

velocity_controller2:
  ros__parameters:
    joints:
      - leg2_joint
    interface_name: velocity
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity

position_controller:
  ros__parameters:
    joints:
      - leg1_joint
      - leg2_joint
    interface_name: position
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity
